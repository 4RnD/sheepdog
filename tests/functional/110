#!/bin/bash

# Test sheep -V/--vnodes

. ./common

function testStartVnodesFailed
{
	( SHEEP_OPTIONS="-V $1" _start_sheep 0 ) && :
	_wait_for_sheep_stop 0
	_cleanup
}

function testStartVnodesSucceeded
{
	SHEEP_OPTIONS="-V $1" _start_sheep 0
	_wait_for_sheep 1
	$DOG node list
	_cleanup
}

function testStartVnodesGatewayFailed
{
	( SHEEP_OPTIONS='-g -V 42' _start_sheep 0 ) && :
	_wait_for_sheep_stop 0
	_cleanup
}

testStartVnodesSucceeded 1
testStartVnodesSucceeded 65535 # UINT16_MAX
testStartVnodesFailed 0
testStartVnodesFailed 65536 # UINT16_MAX + 1
testStartVnodesFailed 65537 # UINT16_MAX + 2
testStartVnodesFailed 4294967297 # UINT32_MAX + 2
testStartVnodesFailed -1
testStartVnodesFailed a
testStartVnodesFailed 42a
testStartVnodesFailed +
testStartVnodesFailed -
testStartVnodesGatewayFailed
