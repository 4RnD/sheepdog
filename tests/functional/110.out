QA output created by 110
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	 1          0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	65535          0
Invalid number of vnodes '0': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '65536': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '65537': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '4294967297': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '-1': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes 'a': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '42a': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '+': must be an integer between 1 and 65535
cannot start sheep 0
Invalid number of vnodes '-': must be an integer between 1 and 65535
cannot start sheep 0
Options '-g' and '-V' can not be both specified
cannot start sheep 0
